addon/utils/plugins/table/tab-input-plugin.ts

```diff
@@ -24,10 +27,67 @@ export default class TableTabInputPlugin implements TabInputPlugin {
         allow: true,
         executor: this.tabHandler as unknown as ManipulationExecutor
       };
+    } else {
+      if(_manipulation.type === 'moveCursorToStartOfElement' || _manipulation.type === 'moveCursorAfterElement') {
+        if(this.isNextElementATable(selection)){
+          return {
+            allow: true,
+            executor: this.selectFirstCell,
+          };
+        }
+      } else if(_manipulation.type === 'moveCursorToEndOfElement' || _manipulation.type === 'moveCursorBeforeElement') {
+        if(this.isPreviousElementATable(selection)){
```

Comment:
this should use the manipulation.element instead of just relying on the selection